packages/@lwc/wire-service/src/wiring.ts

```diff
@@ -233,26 +235,28 @@ export class WireEventTarget {
                 this._cmp[this._wireTarget] = value;
             }
             return false; // canceling signal since we don't want this to propagate
-        } else if ((evt as ComposableEvent).type === 'WireContextEvent') {
-            // NOTE: kill this hack
-            // we should only allow ValueChangedEvent
-            // however, doing so would require adapter to implement machinery
-            // that fire the intended event as DOM event and wrap inside ValueChangedEvent
+        } else if (evt instanceof LinkContextEvent) {
+            const { uid, callback } = evt;
+            // this must dispatch an underlying dom event on the element to connect to
+            // the provider, which uses an internal unique ID defined per adapter, and
+            // provides a callback used as a side channel communication between provider
+            // and consumer adapter.
```

Comment:
i'm having a hard time understanding the comment. Perhaps we can describe the intent, then the purpose of uid and the purpose of callback in separate sentences? ex:
uid - unique adapter id, used to communicate to the provide
callback - enables provider to adapter communication